pkg/meta/base.go

```diff
@@ -2322,17 +2322,8 @@ func (m *baseMeta) checkTrash(parent Ino, trash *Ino) syscall.Errno {
 	}
 	m.Unlock()
 
-	st := m.en.doLookup(Background, TrashInode, name, trash, nil)
-	if st == syscall.ENOENT {
-		next, err := m.en.incrCounter("nextTrash", 1)
-		if err == nil {
-			*trash = TrashInode + Ino(next)
-			attr := Attr{Typ: TypeDirectory, Nlink: 2, Length: 4 << 10, Parent: TrashInode, Full: true}
-			st = m.en.doMknod(Background, TrashInode, name, TypeDirectory, 0555, 0, "", trash, &attr)
-		} else {
-			st = errno(err)
-		}
-	}
+	st := m.ensureTrash(name, trash)
+	go m.preCreateTrash()
```

Comment:
This will double the call of of ensureTrash
I think we can call the preCreateTrash in the last minute of a hour.

We can also cache the trashInode for a hour

Reply:
Those two functions will only be called at most once per hour. Because current trash name is cached in memory, and most fetch requests will be intercepted by the cache.

https://github.com/juicedata/juicefs/blob/7b272d121794105fec74c1a6e2d816e801009785/pkg/meta/base.go#L2317-L2322